src/backend/distributed/utils/metadata_cache.c

```diff
@@ -3222,6 +3384,24 @@ InvalidateEntireDistCache(void)
 }
 
 
+/*
+ * InvalidateDistObjectCache marks all DistObjectCacheHash entries invalid.
+ */
+static void
+InvalidateDistObjectCache(void)
```

Comment:
Should we put a trigger on pg_dist_object that calls a UDF that calls this function? That's what we do for other metadata tables.

Reply:
From what I can tell we track relation changes in `InvalidateDistRelationCacheCallback` which is registered in `InitializeDistCache`

Reply:
That's listening for catalog changes, which we use to invalidate the distributed table cache when someone adds a column to the distributed table. Individual insert/update/delete commands are handled via triggers (e.g. `master_dist_node_cache_invalidate`)